onisite/settings_base.py

```diff
@@ -1,74 +1,215 @@
+# DJANGO CUSTOMIZATIONS
+####################################################################
+# Hosts/domain names valid for this site
+# Should be overridden in settings_local.py, especially for production
+# TODO: Enable overriding by /etc/openoni.ini
+ALLOWED_HOSTS = ['*']
+
+
+CACHES = {
+    'default': {
+        # Development
+        'BACKEND': 'django.core.cache.backends.dummy.DummyCache'
+
+        # Production
+        #'BACKEND': 'django.core.cache.backends.filebased.FileBasedCache',
+        #'LOCATION': '/var/tmp/django_cache',
+        #'TIMEOUT': _ONEWEEK * 8
+    }
+}
+
+
+# Overridden in settings_local.py or /etc/openoni.ini
 DATABASES = {
     'default': {
         'ENGINE': 'django.db.backends.mysql',
         'HOST': 'rdbms',
         'NAME': 'openoni',
         'USER': 'openoni',
         'PASSWORD': 'openoni',
-        }
     }
+}
 
-# Make this unique, and don't share it with anybody.  This MUST be overridden
-# either in settings_local.py or /etc/openoni.ini.
-SECRET_KEY = ''
+# List of configuration classes / app packages in order of priority (i.e., the
+# first item in the list has final say when collisions occur)
+INSTALLED_APPS = (
+    # Default
+#    'django.contrib.admin',
+#    'django.contrib.auth',
+#    'django.contrib.contenttypes',
+    'django.contrib.sessions',
+    'django.contrib.messages',
+    'django.contrib.staticfiles',
 
-# Classes which implement request/response middleware
+    # OpenONI
+    'django.contrib.humanize',  # Added to make data more human-readable
+    'themes.default',
+    'core',
+)
+
+# Suggested order: https://docs.djangoproject.com/en/1.9/ref/middleware/#middleware-ordering
 MIDDLEWARE_CLASSES = (
+    'django.middleware.security.SecurityMiddleware',
+    'core.middleware.TooBusyMiddleware',                               # OpenONI
+    'django.middleware.http.ConditionalGetMiddleware',                 # OpenONI
+    'django.contrib.sessions.middleware.SessionMiddleware',
     'django.middleware.common.CommonMiddleware',
     'django.middleware.csrf.CsrfViewMiddleware',
-    'django.middleware.http.ConditionalGetMiddleware',
-    'core.middleware.TooBusyMiddleware',
+#    'django.contrib.auth.middleware.AuthenticationMiddleware',
+#    'django.contrib.auth.middleware.SessionAuthenticationMiddleware',
+    'django.contrib.messages.middleware.MessageMiddleware',
+    'django.middleware.clickjacking.XFrameOptionsMiddleware',
 )
 
+# Site id differentiates DB data if multiple sites use the same database
+SITE_ID = 1
+
+# Use cache with storing static files; e.g., CSS, images, etc.
+# Development (Default)
+#STATICFILES_STORAGE = 'django.contrib.staticfiles.storage.StaticFilesStorage'
+# Production
+#STATICFILES_STORAGE = 'django.contrib.staticfiles.storage.CachedStaticFilesStorage'
+
+# URL prepended to static file paths; Must include trailing slash
+STATIC_URL = '/media/'
+
+# Directory path to static files
+STATIC_ROOT = os.path.join(BASE_DIR, '.static-media')
```

Comment:
`tmp` in particular makes me think that it would be something which shouldn't stick around for a long time, but really these are the compiled assets the site uses.  Perhaps it should just match whatever path we choose for the `STATIC_URL` also?  so.... `assets` or `static` or something?

Reply:
I'd suggest something clearer than `tmp`, as that could be used for anything at all, and it wouldn't be obvious when it's safe to remove (which is pretty much "never" for compiled assets, except if you are about to re-run the static asset compiler).  I think `compiled-assets` or something makes it pretty clear.

Reply:
Oh, I guess I just repeated @jduss4.  Very nice plagiarism, @jechols!

Reply:
So, now I see what `/static/` is currently being used for. I guess I didn't look very closely earlier. In the [Apache config](https://github.com/open-oni/open-oni/blob/master/conf/openoni.conf#L38) it's used for `robots.txt` and `sitemap.xml`. Only the robots.txt file is covered by the repo. In my local files not tracked by git, the `coverage` library has created some static HTML pages about code coverage. This must be done during the initial install?

Having `robots.txt` and `sitemap.xml` not at the root of the domain is fairly uncommon, so I think that may be another reason to change things here.

For `STATIC_URL`, I'm still leaning towards `/static/` or maybe even just at `/` so they're served from `https://(hostname)/js/` instead of a sub-URI. The latter is the Django default, from how I understand the documentation. It also depends on whether or not we want to apply Apache directives to the static files, as is currently done where Apache adds caching headers. I think it'd be simpler to handle all of that inside Django though.

For `STATIC_ROOT` where it stores the files, if we don't nest it inside a `cache/` (or similarly named) directory as just `static/`, I'd go with something like `static-compiled/` or `static-cached/`? `static-compiled` is hopefully clearer in it's purpose and communicates that it can't simply be deleted like other `tmp/` and `cache/` directories. :thinking:

Reply:
I'm thinking `/static/compiled` then other static files and directories could be placed in the parent /static folder. I agree that this more semantically identifies the purpose than "media" has.

Reply:
I'm going to leave this as is for now but will keep this in mind for when working on Docker deployment etc. Changing this also requires modifying Apache config, which is going to need an overhaul later anyway.

Reply:
I take that back. I have to update the Apache config anyway for the STATIC_URL change, so I'm changing this to `/static/compiled/` for now